### PR TITLE
Fix for python 3.12: replace removed `distutils`

### PR DESCRIPTION
**Pull request by vladsavelyev**
`distutils` was fully removed in Python 3.12: https://peps.python.org/pep-0632/#migration-advice

1. Replace `distutils.version.StrictVersion` with `packaging.version`
2. Replace `distutils.util.strtobool` with re-implementation
3. Replace `distutils.dir_util.copy_tree` with `shutil.copytree` calls

### CODE REVIEW

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

File: multiqc/modules/picard/picard.py [hunk 1]
```diff
@@ -157,3 +157,25 @@ def multiply_hundred(self, val):
         except ValueError:
             pass
         return val
+
+
+def strtobool(val):
```
Comment (ewels) [inline multiqc/modules/picard/picard.py:162]: Better to put this in the MultiQC utils, rather than having it in a module?

**Review by ewels — APPROVED** [review]
Nice 👍🏻

**Comment by ewels** [comment]
@multiqc-bot fix linting

File: multiqc/modules/picard/picard.py (hunk 1 above)
Comment (vladsavelyev) [reply multiqc/modules/picard/picard.py:162]: It think it makes sense to move it if/when it's going to be used by other modules

Comment (vladsavelyev) [reply multiqc/modules/picard/picard.py]: Though it will actually fix the test, so doing it now 😅

**Comment by vladsavelyev** [comment]
(Please merge https://github.com/ewels/MultiQC/pull/2115 first)

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

**Comment by vladsavelyev** [comment]
Sigh, `dirs_exist_ok` to `shutil.copytree()` was only added in Py 3.8 :( https://docs.python.org/3/whatsnew/3.8.html#shutil

Should probably add an if-else to import distutils for 3.6 and 3.7 instead of calling shutils.

**Comment by ewels** [comment]
If we do, maybe we add a note to https://github.com/ewels/MultiQC/issues/2039

It would be really nice to know what Python versions people are using! No idea how many people are still on 3.7..

**Comment by vladsavelyev** [comment]
Good point. I thought we are dropping only 3.6; but if we drop 3.7 as well, it should be nice.

**Review by ewels — APPROVED** [review]
Great!

**Pull request merged**